### PR TITLE
Move the SOF app from zephyr to where it belongs

### PR DESCRIPTION
**Pull request by nashif**
This moves (not deleted in zephyr yet) the SOF app into SOF, where it belongs.
No need for duplicated overlays and one place to build everything.

You now use scripts or west directly to build this and able to make all changes needed in the configuration files without having go via zpehyr or go through the complexity of a second layer of configurations (overlays)

- zephyr: use zephyr prefix for includes
- zephyr: app: move main SOF app from zephyr samples
- zephyr: module: point to where the app is
- app: zephyr: remove overlays and adapt script

### CODE REVIEW

**Review by aborisovich — CHANGES_REQUESTED** [review]
Quick review, two minor python issues.
Going to test Windows build in a minute,
As I understand the idea we are going to use overlays as common for everyone?
I am not sure this is right.
Yes we can do this for debug_overlay.conf but it does not make sense to use IPC4 overlay with anothing older than TGL - it probably won't even build and nobody did it before.
I would go back to old approach with platform names, something like this:
image
Then we can see which overlays are what for.
One more thing - I would change a name from "app" to "cSOF" maybe?
Like "Converged SOF" so it matches our official name :-)

File: scripts/xtensa-build-zephyr.py [hunk 1]
```diff
@@ -483,28 +483,23 @@ def build_platforms():
...
+			overlays.append(str(pathlib.Path(SOF_TOP, "app", "debug_overlay.conf")))
 
 		# The '-i IPC4' is a shortcut for '-o path_to_ipc4_overlay', we are good if both
 		# are provided, because it's no harm to merge the same overlay twice.
 		if args.ipc == "IPC4":
-			overlays.append(str(pathlib.Path(SOF_TOP, "overlays", platform, platform_dict["IPC4_CONFIG_OVERLAY"])))
+			overlays.append(str(pathlib.Path(SOF_TOP, "app", platform, platform_dict["IPC4_CONFIG_OVERLAY"])))
 
 		overlays = ";".join(overlays)
 		build_cmd.append(f"-DOVERLAY_CONFIG={overlays}")
```
[the remark this answers is not in the document]
Comment (nashif) [reply scripts/xtensa-build-zephyr.py:505]: feature overlays are still available, we still need that. The IPC4 overlay was moved to the app.

**Comment by marc-hb** [comment]
Thanks Anas for helping with this, much appreciated.

> @marc-hb @keqiaozhang @wszypelt does anything need to be done for CI to use this update ?


Sorry I don't know why the Zephyr build was successful in https://github.com/thesofproject/sof/runs/7322499212 but failed in https://sof-ci.01.org/sofpr/PR6004/build891/build (error below). The changes to `sof/scripts/xtensa-build-zephyr.py` are small and simple; they look good to me.

```
SOF_TOP=/srv/home/jenkins/workspace/sof_generic_build
west_top=/srv/home/jenkins/workspace
In dir: /srv/home/jenkins/workspace; running command: west status hal_xtensa sof
FATAL ERROR: file not found: /srv/home/jenkins/workspace/sof_generic_build/west.yml
```


@keqiaozhang, @XiaoyunWu6666 ?


Jenkins does a bit more with `sof/scripts/xtensa-build-zephyr.py`, I found other suspicious errors in jenkins job 3667:

```
07:05:26 Traceback (most recent call last):
07:05:26   File "scripts/xtensa-build-zephyr.py", line 631, in <module>
07:05:26     main()
07:05:26   File "scripts/xtensa-build-zephyr.py", line 624, in main
07:05:26     run_clone_mode()
07:05:26   File "scripts/xtensa-build-zephyr.py", line 594, in run_clone_mode
07:05:26     raise RuntimeError("Zephyr found already! Not downloading it again")
07:05:26 RuntimeError: Zephyr found already! Not downloading it again
07:05:26 fatal: cannot change to 'zephyrproject/zephyr': No such file or directory
```

File: scripts/xtensa-build-zephyr.py [hunk 2]
```diff
@@ -483,31 +483,28 @@ def build_platforms():
 		build_cmd = ["west"]
```
Comment (aborisovich) [inline scripts/xtensa-build-zephyr.py:483]: You added configuration for MTL board, please add missing python dictionary so we can build it too:
	{
		"name": "mtl",
		"PLAT_CONFIG": "intel_adsp_ace15_mtpm",
		"XTENSA_CORE": "ace10_LX7HiFi4_RI_2020_5",
		"XTENSA_TOOLS_VERSION": f"RI-2020.5{xtensa_tools_version_postfix}",
		"RIMAGE_KEY": pathlib.Path("modules", "audio", "sof", "keys", "otc_private_key_3k.pem
	},

**Review by aborisovich — CHANGES_REQUESTED** [review]
Sorry for delays in testing Windows build (i need to configure my environment).
I'll let you know as soon as its verified.
Meanwhile found one more thing to python (enable MTL build).

**Comment by keqiaozhang** [comment]
SOFCI TEST

**Comment by keqiaozhang** [comment]
SOFCI TEST

**Comment by keqiaozhang** [comment]
> Jenkins does a bit more with `sof/scripts/xtensa-build-zephyr.py`, I found other suspicious errors in jenkins job 3667:

@marc-hb , this build failure is not caused by this PR. I checked the Jenkins log and found that our [sofbld10](https://sof-jenkins.ostc.intel.com/computer/sofbld10) failed to clone zephyrporject, because there's `.west` folder in `/srv/home/jenkins/workspace/`. I have deleted this folder and zephyr build passed.

**Comment by aborisovich** [comment]
> Sorry for delays in testing Windows build (i need to configure my environment). I'll let you know as soon as its verified. Meanwhile found one more thing to python (enable MTL build).

Confirmed Windows build working (built TGL successfully).

**Comment by aborisovich** [comment]
> 

It may be my fault as I'm messing around with processes in [PR6005](https://github.com/thesofproject/sof/pull/6005), sorry.

File: scripts/xtensa-build-zephyr.py (hunk 2 above)
Comment (nashif) [reply scripts/xtensa-build-zephyr.py:483]: I will probably remove it and leave it to you to add this alongside other mtl related changes.

**Comment by marc-hb** [comment]
> It may be my fault as I'm messing around with processes in https://github.com/thesofproject/sof/pull/6005, sorry.

Thanks @aborisovich , very good catch I didn't realize 6005 could break CI _even before getting merged_ but because of big west differences then yes of course it can! Ideally it shouldn't but in practice it can because our workers clean but don't clone everything from scratch. I prefixed the Github PR title with `[SKIP CI]` to stop sof-jenkins from looking at it. This affects `sof-jenkins` CI ONLY, ignored by everything else. You do not need to change any commit message or anything.

**Review by marc-hb — APPROVED** [review]
https://sof-ci.01.org/sofpr/PR6004/build896/devicetest/ is all green expect one unrelated and unfortunately common suspend/resume failure.

The Python, CMake and C changes LGTM, thanks!

I don't know about the overlays and .conf files but I blindly trust @nashif with that :-)

**Comment by lgirdwood** [comment]
@aborisovich good for you now ?

**Review by aborisovich — APPROVED** [review]
Never trust anyone blindly, we are all developers ;-)
I just wanted to remove boards/intel_adsp_ace15_mtpm.conf from this PR just for consistency (or add it to python so it is buildable). But maybe it does not matter that much.
Everything else looks perfect, windows build works too, approving.

**Comment by nashif** [comment]
> I just wanted to remove boards/intel_adsp_ace15_mtpm.conf from this PR just for consistency (or add it to python so it is buildable).

removed

**Comment by wszypelt** [comment]
@lgirdwood 
It seems to me that everything should be working properly now, I have made retests for this and other PR

**Comment by lgirdwood** [comment]
> @lgirdwood It seems to me that everything should be working properly now, I have made retests for this and other PR

Thanks for checking @wszypelt - I will rerun as the CI error log is still not loading for me.

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by marc-hb** [comment]
@lgirdwood the magic comment "SOFCI TEST" is scanned by SOFCI TEST *only*, QB does not care.

I looked at the internal 9877300 logs for https://sof-ci.01.org/sof-pr-viewer/#/build/PR6004/build9877263 and this can merged. I confirm that the build in QB was successful, only the deployment to 2 devices (TGL and WHL) failed:
```
pytest tests/sof/fw_00_basic/test_00_load_fw_basic.py::TestLoadFwBasic   --fw="C:\Testing\Builds\pull-6004-merge_740929\CNL\sof-cnl.ri" --ldc-file="C:\Testing\Builds\pull-6004-merge_740929\CNL\sof-cnl.ldc" --zabbix --tests-startup-label='SMOKE_pull-6004-merge_740929' ...
...
04:34:51,363 INFO  - INTERNALERROR>   File "/usr/local/lib/python3.6/dist-packages/execnet/gateway_socket.py", line 91, in create_io
04:34:51,363 INFO  - INTERNALERROR>     sock.connect((host, port))
04:34:51,363 INFO  - INTERNALERROR> ConnectionRefusedError: [Errno 111] Connection refused
```
I doubt this PR even affects QB at all.

(https://sof-ci.01.org/sofpr/PR6004/build906/devicetest/ is all green except for one unfortunately common suspend/resume failure.)

**Pull request merged**